forge/src/main/java/org/openjdk/skara/forge/gitlab/GitLabMergeRequest.java

```diff
@@ -646,28 +646,28 @@ public void removeLabel(String label) {
         request.put("")
                .body("labels", String.join(",", labels))
                .execute();
+        this.labels = List.copyOf(labels);
     }
 
     @Override
     public void setLabels(List<String> labels) {
         request.put("")
                .body("labels", String.join(",", labels))
                .execute();
-        this.labels = labels.stream().map(this::labelNameToLabel).collect(Collectors.toList());
+        this.labels = List.copyOf(labels);
     }
 
     @Override
     public List<Label> labels() {
-        if (labels == null) {
-            var currentJson = request.get("").execute().asObject();
-            labels = currentJson.get("labels")
-                                .stream()
-                                .map(s -> labelNameToLabel(s.asString()))
-                                // Avoid throwing NPE for unknown labels
-                                .filter(Objects::nonNull)
-                                .sorted()
-                                .collect(Collectors.toList());
-        }
+        return labels.stream()
+                .map(this::labelNameToLabel)
+                // Avoid throwing NPE for unknown labels
+                .filter(Objects::nonNull)
+                .toList();
```

Comment:
You're losing `.sorted()`. Maybe it doesn't matter?

Reply:
Hm, probably shouldn't matter, but I retained the sorted property on the `labels` field in the constructor, so should probably make sure all assignments to that field are made with sorted lists for consistency.

Reply:
Sounds good. In general I think it’s best to keep things sorted, in order to minimize spurious changes in places that might end up being user visible.